### PR TITLE
Feature/333/check in history

### PR DESCRIPTION
**Pull request by kcinay055679**

### CODE REVIEW

File: frontend/src/app/app.module.ts [hunk 1]
```diff
@@ -79,6 +81,8 @@ export const MY_FORMATS = {
     KeyresultComponent,
     ApplicationHeaderComponent,
     ConfidenceComponent,
+    CheckInHistoryDialogComponent,
+    CheckInHistoryDialogComponent,
```
Comment (peggimann) [inline frontend/src/app/app.module.ts:85]: if twice remove once

File: frontend/src/app/shared/dialog/check-in-history-dialog/check-in-history-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      <div class="d-flex flex-row flex-wrap">
+        <span class="me-1">{{ checkIn.createdOn | date: "dd.MM.yyyy" }}</span>
+        <span>-</span>
+        <div class="ms-1 me-1" ngSwitch="{{ checkIn.valueMetric }}">
+          <span *ngSwitchCase="undefined">Zone: {{ checkIn.zone }};</span>
+          <span *ngSwitchDefault>ValueMetric: {{ checkIn.valueMetric }};</span>
+        </div>
+        <span>Confidence: {{ checkIn.confidence }} / 10;</span>
+      </div>
+      <p *ngIf="checkIn.changeInfo !== ''">Veränderungen:{{ checkIn.changeInfo }}</p>
```
Comment (peggimann) [inline frontend/src/app/shared/dialog/check-in-history-dialog/check-in-history-dialog.component.html:19]: add whitespaces

File: frontend/src/app/shared/dialog/check-in-history-dialog/check-in-history-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  <div mat-dialog-title class="d-flex flex-row justify-content-start mt-2">
+    <h1 class="d-flex justify-content-start title w-50 mt-2">Check-in History</h1>
+    <div class="d-flex justify-content-end w-50">
+      <mat-icon class="icon mt-2 me-0" mat-dialog-close>close</mat-icon>
+    </div>
+  </div>
+  <mat-dialog-content class="d-flex justify-content-start flex-column ms-3 mb-2 mt-3 gap-4">
+    <div *ngFor="let checkIn of checkInHistory | async; index as i">
+      <div class="d-flex flex-row flex-wrap">
+        <span class="me-1">{{ checkIn.createdOn | date: "dd.MM.yyyy" }}</span>
```
Comment (peggimann) [inline frontend/src/app/shared/dialog/check-in-history-dialog/check-in-history-dialog.component.html:11]: add to do

File: frontend/src/app/shared/services/check-in.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+describe('CheckInService', () => {
+  let service: CheckInService;
+
+  const testedKeyResultId: number = 8;
+  const firstCheckIn: CheckInMin = {
+    valueMetric: 77,
+    confidence: 5,
+    zone: undefined,
+    changeInfo: '',
```
Comment (peggimann) [inline frontend/src/app/shared/services/check-in.service.spec.ts:15]: move to /shared/testData.ts

**Review by peggimann — CHANGES_REQUESTED** [review]

File: frontend/src/app/app.module.ts (hunk 1 above)
Comment (MarcEgliP) [reply frontend/src/app/app.module.ts:85]: removed

File: frontend/src/app/shared/dialog/check-in-history-dialog/check-in-history-dialog.component.html (hunk 2 above)
Comment (MarcEgliP) [reply frontend/src/app/shared/dialog/check-in-history-dialog/check-in-history-dialog.component.html:19]: Added

File: frontend/src/app/shared/services/check-in.service.spec.ts (hunk 4 above)
Comment (MarcEgliP) [reply frontend/src/app/shared/services/check-in.service.spec.ts:15]: Moved

**Review by peggimann — APPROVED** [review]

**Pull request merged**